### ISSUE TITLE
Raspberry 4 Pi Buster

### ISSUE THREAD

**Issue by gerryb** [pothosware/SoapyFCDPP#18]
Fails to  connect

2024-07-29 23:49:12,445 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - Started sdr source: soapy_connector -s 192000 -f 28074000 -p 60261 -c 59449 -d driver=fcdpp
2024-07-29 23:49:12,456 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: setting up control socket...
2024-07-29 23:49:12,456 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: control socket started on 59449
2024-07-29 23:49:12,457 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: socket setup complete, waiting for connections
2024-07-29 23:49:12,545 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: [INFO] makeFCDPP
2024-07-29 23:49:12,545 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: hid_open_path failed to open: 1-1.4:1.2
2024-07-29 23:49:12,546 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: Connector::open() failed
2024-07-29 23:49:30,765 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - Started sdr source: soapy_connector -s 192000 -f 28074000 -p 60261 -c 59449 -d driver=fcdpp
2024-07-29 23:49:30,777 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: setting up control socket...
2024-07-29 23:49:30,777 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: control socket started on 59449
2024-07-29 23:49:30,777 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - INFO - STDOUT: socket setup complete, waiting for connections
2024-07-29 23:49:30,854 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: [INFO] makeFCDPP
2024-07-29 23:49:30,855 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: hid_open_path failed to open: 1-1.4:1.2
2024-07-29 23:49:30,855 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - WARNING - STDERR: Connector::open() failed
2024-07-29 23:49:30,867 - owrx.source.ffa774a3-e9a6-48f2-ab25-1c0410cb7210 - ERROR - Exception during postStart()
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/owrx/source/__init__.py", line 427, in start
    self.postStart()
  File "/usr/lib/python3/dist-packages/owrx/source/connector.py", line 62, in postStart
    self.controlSocket.connect(("localhost", self.controlPort))
ConnectionRefusedError: [Errno 111] Connection refused

**Comment by phlash** [comment]
Hi @gerryb - this may be a permissions problem with the USB device - I'll check if I can reproduce here on my Pi, meanwhile can you check that your OpenwebRX user has access to the FCD device node under `/dev/bus/usb/001/...` please :grin:

**Comment by gerryb** [comment]
All there is 

 

***@***.***:/dev/bus/usb# ls

001  002

***@***.***:/dev/bus/usb#

 

***@***.***:/dev/bus/usb/001# ls

001  002  004  005  006    These are not files nor directories…

 

Thank for your help and maintaining this

My Fucube Pro is not quite old but better on HF thant many other dongles… So would be great on my RaspberryPi

 

Gerry

**Comment by phlash** [comment]
Ah sorry @gerryb, I should have been more specific: with the FCDPP plugged in, please look at the details of the USB device nodes and identify which is the FCDPP as follows:
```
ls -al /dev/bus/usb/*
lsusb
```
this will show the owner and group of each device node, along with the access permissions, then which device node is the FCDPP. The user attempting to open the FCDPP (in your case the user that OpenwebRX runs as, normally `openwebrx`) must have read and write permission to the device node. If you see all files with owner and group `root:root` and permissions `crw-rw-r--` it's likely that nothing is setting the permissions you need.

Depending on what software you have installed for the FCDPP, there may be a rules file to set the permissions when the FCDPP is plugged in, check in `/etc/udev/rules.d` for files with `funcube` or `fcd` in the name. If you don't see anything, then create one called `99-funcube-dongle.rules` and paste these rules in:
```
# Udev rules for the Funcube Dongle Pro (fb56) & Pro+ (0xfb31)

# HIDAPI/libusb:
SUBSYSTEMS=="usb", ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="fb56", GROUP:="users", MODE:="0666", SYMLINK+="FCD"
SUBSYSTEMS=="usb", ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="fb31", GROUP:="users", MODE:="0666", SYMLINK+="FCDPP"
```
re-plugging the FCDPP should now fix the permissions.

Good luck :)

**Comment by gerryb** [comment]
Fixed.   thanks

 

/dev/bus/usb/002:

total 0

drwxr-xr-x 2 root root       60 Jan  1  1970 .

drwxr-xr-x 4 root root       80 Jan  1  1970 ..

crw-rw-r-- 1 root root 189, 128 Aug  3 15:57 001

 

I saw the broad correct permissions before but forgot to pass it on

 

There were no rules 99-funcube-dongle.rules

Adding them fixed the issue.

 

Thanks

 

Next getting my Airspy mini to weork on OpenWebrx, that is is for elsewhere.

 

Thank you for this

 

Gerard Bulger
